### PR TITLE
(SUP-3564) Add support for PE replica

### PR DESCRIPTION
**Pull request by taikaa**

### CODE REVIEW

**Review by m0dular — APPROVED** [review]
This is mostly good to go, but I need to make some additional changes before we merge and release.  I'll approve this but add the do not merge label until everything is ready to go.

**Comment by taikaa** [comment]
Great. Thank you @m0dular!

**Comment by m0dular** [comment]
I opened #89 that contains your commit, so we just merge that PR  in lieu of this one.

**Pull request closed without merging**
